### PR TITLE
to_html formatter not called for float values in a mixed-type column

### LINKED ISSUE #13021: to_html formatter not called for float values in a mixed-type column
#### When column x has float64 type, the custom formatter is applied to every value

```
t = pd.DataFrame([[0], [10.4], [3]], columns=['x'])
display(HTML(t.to_html(formatters=dict(x='xxx'.format))))

   x
------
0  xxx
1  xxx
2  xxx
```
#### But when column x is mixed type, the formatter isn't called for (only) floating point values

```
t = pd.DataFrame([['a'], [0], [10.4], [3]], columns=['x'])
display(HTML(t.to_html(formatters=dict(x='xxx'.format))))

   x
------
0  xxx
1  xxx
2  10.4
3  xxx
```
## INSTALLED VERSIONS

commit: None
python: 2.7.11.final.0
python-bits: 64
OS: Darwin
OS-release: 14.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0
nose: None
pip: 1.5.4
setuptools: 20.7.0
Cython: None
numpy: 1.11.0
scipy: 0.17.0
statsmodels: None
xarray: None
IPython: 4.1.2
sphinx: None
patsy: None
dateutil: 2.4.2
pytz: 2016.3
blosc: None
bottleneck: None
tables: 3.2.2
numexpr: 2.5.2
matplotlib: 1.5.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: None
boto: None

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [x] closes #13021
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

can't shortcut completely without changing existing tested behaviour in a couple of tests.

for instance, the expected output for `test_to_string_with_formatters` in `pandas\tests\io\formats\test_format.py` has two spaces between float and object column, whereas shortcuting would only have one (the same as between the int and float columns)

```
  int  float    object
0 0x1 [ 1.0]  -(1, 2)-
1 0x2 [ 2.0]    -True-
2 0x3 [ 3.0]   -False-
```
if it's OK to change a couple of tests, we can remove this subtle leading_space bug in this PR.

### CODE REVIEW

File: pandas/tests/io/formats/test_to_html.py [hunk 1]
```diff
@@ -633,3 +633,13 @@ def test_to_html_invalid_classes_type(classes):
 
     with pytest.raises(TypeError, match=msg):
         df.to_html(classes=classes)
+
+
+def test_to_html_formatters_object_type():
+    # GH 13021
+    def f(x):
+        return x if type(x) is str else '${:,.0f}'.format(x)
```
Comment (WillAyd) [inline pandas/tests/io/formats/test_to_html.py]: Use isinstance instead of type

File: pandas/tests/io/formats/test_to_html.py [hunk 2]
```diff
@@ -633,3 +633,13 @@ def test_to_html_invalid_classes_type(classes):
...
+
+    df = pd.DataFrame([['a'], [0], [10.4], [3]], columns=['x'])
+    result = df.to_html(formatters=dict(x=f))
+    assert '$10' in result
```
Comment (WillAyd) [inline pandas/tests/io/formats/test_to_html.py]: Can we make a better assertion here? I understand the original issue pertained to floats only but probably doesn't hurt to make an assertion around the other values here as well

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, ex @WillAyd comments, this fully replicates the original issue?

**Comment by simonjayhawkins** [comment]
>  this fully replicates the original issue?

yes.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by simonjayhawkins** [comment]
green. but not all checks ran. needs a restart?

**Comment by jreback** [comment]
hmm yeah, merge master and push again.

**Comment by simonjayhawkins** [comment]
gbq failure, otherwise good.

**Comment by jreback** [comment]
yeah working on fixing gbq now

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Comment by jreback** [comment]
I reverted this: https://github.com/pandas-dev/pandas/commit/fc5f292377412fa7bc824d2c2cc69f2ae45bd7f9 as CI is failing

**Pull request merged**